sympy/polys/matrices/domainmatrix.py

```diff
@@ -1877,33 +1871,45 @@ def rref(self):
         >>> rref_pivots
         (0, 1, 2)
 
-        See Also
-        ========
+        Parameters
+        ==========
 
-        convert_to, lu
-        rref_den
+        method : str, optional (default: 'auto')
+            The method to use to compute the RREF. The default is ``'auto'``,
+            which will attempt to choose the fastest method. The other options
+            are ``'GJ'``, which uses Gauss-Jordan elimination, ``'FF'``, which
+            uses fast fraction-free elimination, and ``'CD'``, which will
+            clear denominators before using Gauss-Jordan elimination. The
+            domain of the returned matrix will always be in the associated
+            field.
 
-        """
-        if not self.domain.is_Field:
-            raise DMNotAField('Not a field')
-        rref_ddm, pivots = self.rep.rref()
-        return self.from_rep(rref_ddm), tuple(pivots)
-
-    def rref_den(self):
-        r"""
-        Returns reduced-row echelon form with denominator and list of pivots.
+            The sparse implementations are always used by default but if the
+            method is e.g. ``'GJ_dense'`` then the dense implementation will
+            be used. The returned matrix will always have the same format
+            (sparse or dense) as the input.
 
         Returns
         =======
 
         (DomainMatrix, list)
             reduced-row echelon form and list of pivots for the DomainMatrix
 
-        Raises
-        ======
+        See Also
+        ========
 
-        ValueError
-            If the domain of DomainMatrix not a Field
+        rref_den
+            RREF with denominator
+        sympy.polys.matrices.rref.dm_rref
+            The function that implements this.
+
+        """
+        return dm_rref(self, method=method)
```

Comment:
My opinion is that pulling out the object methods like this, is not very nice,
and especially it is starting to make documentation more messier.
I wondered just refactoring around the `den` parameter or simplifying the logic for `dense/sparse` option can pull things back here.

Reply:
I have tried to simplify this code but I pushed the change in gh-25443 rather than here because that is the PR where I want those things to be merged. This PR is for considering the use of `DomainMatrix.inv` for arbitrary matrices.